openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -194,8 +194,14 @@ def award_program_certificates(self, username):
                 )
             except Exception:  # pylint: disable=broad-except
                 # keep trying to award other certs, but retry the whole task to fix any missing entries
-                LOGGER.exception('Failed to award certificate for program %s to user %s', program_uuid, username)
-                retry = True
+                error_message = 'Failed to award certificate for program {uuid} to user {username}.'.format(
+                    uuid=program_uuid, username=username)
```

Comment:
Lets call this `warning_message` and move it inside if block as it is the only place it is used

Reply:
I will change it, also this is used in else block as well.